database/integration_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package database
+
+import (
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-vela/server/database/worker"
+
+	"github.com/go-vela/types/library"
+)
+
+func TestDatabase_Integration(t *testing.T) {
+	// check if we should skip the integration test
+	if os.Getenv("INTEGRATION") == "" {
+		t.Skipf("skipping %s integration test due to environment variable constraint", t.Name())
+	}
+
+	// setup tests
+	tests := []struct {
+		failure bool
+		name    string
+		config  *config
+	}{
+		{
+			name:    "success with postgres",
+			failure: false,
+			config: &config{
+				Driver:           "postgres",
+				Address:          "postgres://vela:notARealPassword12345@localhost:5432/vela",
+				CompressionLevel: 3,
+				ConnectionLife:   10 * time.Second,
+				ConnectionIdle:   5,
+				ConnectionOpen:   20,
+				EncryptionKey:    "A1B2C3D4E5G6H7I8J9K0LMNOPQRSTUVW",
+				SkipCreation:     false,
+			},
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			db, err := New(
+				WithAddress(test.config.Address),
+				WithCompressionLevel(test.config.CompressionLevel),
+				WithConnectionLife(test.config.ConnectionLife),
+				WithConnectionIdle(test.config.ConnectionIdle),
+				WithConnectionOpen(test.config.ConnectionOpen),
+				WithDriver(test.config.Driver),
+				WithEncryptionKey(test.config.EncryptionKey),
+				WithSkipCreation(test.config.SkipCreation),
+			)
+			if err != nil {
+				t.Errorf("unable to create new database engine for %s: %v", test.name, err)
+			}
+
+			driver := db.Driver()
+			if !strings.EqualFold(driver, test.config.Driver) {
+				t.Errorf("Driver() is %v, want %v", driver, test.config.Driver)
+			}
+
+			err = db.Ping()
+			if err != nil {
+				t.Errorf("unable to ping database engine for %s: %v", test.name, err)
+			}
+
+			t.Run("test_workers", func(t *testing.T) {
+				testWorkers(t, db)
+			})
+
+			err = db.Close()
+			if err != nil {
+				t.Errorf("unable to close database engine for %s: %v", test.name, err)
+			}
+		})
+	}
+}
+
+func testWorkers(t *testing.T, db Interface) {
+	// used to track the number of methods we call for workers
+	counter := 0
```

Comment:
This should start at 2 instead of subtracting 2 below. (With an appropriate comment)